### PR TITLE
add operator gather for jax frontend

### PR DESCRIPTION
**Pull request by keawchot**

### CODE REVIEW

**Comment by keawchot** [comment]
- [] #4095

**Comment by sharique1006** [comment]
Hi @keawchot, the arguments in the frontend should be similar to https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.gather.html and you should accordingly implement the function compositionally.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by keawchot** [comment]
I would like to continue working on this PR.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
